src/MainWindow.cpp

```diff
@@ -4562,6 +4578,52 @@ void MainWindow::setupViews()
     showHideViewColumnsForAccountType(currentAccountType);
 }
 
+bool MainWindow::getShowNoteThumbnails() const {
+    QVariant showThumbnails = getApplicationSetting(
+        *m_pAccount, QUENTIER_UI_SETTINGS, LOOK_AND_FEEL_SETTINGS_GROUP_NAME,
+        SHOW_NOTE_THUMBNAILS_SETTINGS_KEY,
+        QVariant::fromValue(DEFAULT_SHOW_NOTE_THUMBNAILS));
```

Comment:
I'm sorry but I must confess, I don't really like this :disappointed: 
I believe I understand your motivation behind doing this - you tried to avoid repeating similar code. But that code was, well, easily understandable. Since C++ doesn't have named parameters feature, the more parameters you need to pass with each call, the harder it is to read such code and getting/setting stuff from/to app settings is a frequent enough activity.

Just for the sake of comparison, here's how these four lines would look written "the old way":
```
ApplicationSettings appSettings(*m_pAccount, QUENTIER_UI_SETTINGS);                                                     
appSettings.beginGroup(LOOK_AND_FEEL_SETTINGS_GROUP_NAME);                                                              
QVariant hideThumbnailsFor = appSettings.value(HIDE_NOTE_THUMBNAILS_FOR_SETTINGS_KEY);                                  
appSettings.endGroup(); 
```
The number of lines is precisely the same as with `getApplicationSetting` but to me it looks easier to grasp because the parameters used are split across some different small actions - create `appSettings` with `*m_pAccount` and `QUENTIER_UI_SETTINGS`, begin group with a particular group name and get the value for a particular key. Plain and simple.

Sometimes it's hard to avoid the structure with methods accepting many parameters - there are many such methods in e.g. `LocalStorageManager` in libquentier. Even there it might be possible to group some common settings together in some class/struct and pass them as a single parameter for the sake of simpler interfacing. If I ever manage to get to this in future, I might actually do it. But again, accessing stuff within `ApplicationSettings` doesn't seem to have to be that hard.

So I'm sorry but I kindly ask you to return to the conventional way of getting/setting persistent app settings.